src/lib/common-yargs.js

```diff
@@ -0,0 +1,191 @@
+const fs = require('fs')
+const parseGitHubURL = require('./parse-github-url')
+
+const withToken = options => yargs =>
+	yargs
+		.option(
+			'token',
+			Object.assign(
+				{},
+				{
+					alias: 't',
+					describe: 'GitHub personal access token. \nGenerate one at https://github.com/settings/tokens.',
+					type: 'string',
+				},
+				options
+			)
+		)
+		.check(argv => {
+			if (!argv.token && !process.env.GITHUB_PERSONAL_ACCESS_TOKEN) {
+				throw new Error('Missing required argument: token.')
+			}
+			return true
+		})
+		.default('token', process.env.GITHUB_PERSONAL_ACCESS_TOKEN, '$GITHUB_PERSONAL_ACCESS_TOKEN')
+
+const withJson = options => yargs =>
+	yargs.option(
+		'json',
+		Object.assign(
+			{},
+			{
+				alias: 'j',
+				describe: 'Format command output as a JSON-formatted string',
+				type: 'boolean',
+			},
+			options
+		)
+	)
+
+const withBody = options => yargs =>
+	yargs
+		.option(
+			'body',
+			Object.assign(
+				{},
+				{
+					alias: 'b',
+					describe: 'Path to pull request body',
+					type: 'string',
+				},
+				options
+			)
+		)
+		.check(argv => {
+			/**
+			 * Sometimes (especially during testing), `process.stdin.isTTY` gives a false negative.
+			 * So make sure that '/dev/stdin' can be read. If not, then consider `stdin` to be absent.
+			 */
+			let hasStdin
```

Comment:
I am not sure this works?

```
cat example.txt | node bin/github.js create-comment https://github.com/Financial-Times/github/pull/49
Error with GitHub network request/response. [{"resource":"IssueComment","code":"unprocessable","field":"data","message":"Body cannot be blank"}]
```

Reply:
Ah. I'll sort that in another PR, cheers